### PR TITLE
graphgist search

### PR DESCRIPTION
**Pull request by whatSocks**

### CODE REVIEW

**Comment by whatSocks** [comment]
demo https://www.loom.com/share/29071290457943ff9ea916c6a1678d42

File: api/src/schema.graphql [hunk 1]
```diff
@@ -1,18 +1,39 @@
...
+    WHERE toLower(category.name) CONTAINS toLower($searchString)
+    RETURN asset
+    UNION
+    MATCH (asset:GraphGist {status:'live'})<-[:WROTE]-(author:Person)
+    WHERE toLower(author.name) CONTAINS toLower($searchString)
+    RETURN asset
+  """)
+  graphGistsByCategory(slug: String): [GraphGist] @cypher(statement: """
+    MATCH (asset:GraphGist {status:'live'})-[:FOR_INDUSTRY|:FOR_USE_CASE]->(category)
+    WHERE category.slug = $slug
```
Comment (jexp) [inline api/src/schema.graphql:17]: full db scan, better use a union

File: api/src/schema.graphql [hunk 2]
```diff
@@ -1,18 +1,39 @@
+type Query {
+  graphGistSearch(searchString: String): [GraphGist] @cypher(statement: """
+    MATCH (asset:GraphGist {status:'live'})
+    WHERE toLower(asset.title) CONTAINS toLower($searchString)
```
Comment (jexp) [inline api/src/schema.graphql:4]: in the future we can use an fulltext index here

**Comment by jexp** [comment]
Oh you should not check in the .env file, that should be handled by secrets in the deployment.

File: docker-compose.yml [hunk 3]
```diff
@@ -18,9 +18,6 @@ services:
     ports:
       - 4001:4001
     environment:
-      - NEO4J_URI=bolt://192.168.99.13:7687
```
Comment (jexp) [inline docker-compose.yml:21]: we should still be able to use docker-compose for testing ...

**Pull request merged**